File: golang_solutions/day01/day01.go
package day01

import (
	"log"
	"os"
	"strconv"
	"strings"
)

func common(filename string) []string {

	// Read input file

	input, err := os.ReadFile(filename)

	if err != nil {
		log.Fatal("File not found")
	}

	lines := strings.Split(string(input), "\n")

	return lines
}

func Part1(filename string) int {

	lines := common(filename)

	sum := 0
	for _, line := range lines {

		chars := strings.Split(line, "")

		numbers := []int{}
		for _, char := range chars {
			number, err := strconv.Atoi(char)

			if err != nil {
				continue
			}

			numbers = append(numbers, number)

		}

		sum = sum + 10*numbers[0] + numbers[len(numbers)-1]
	}

	return sum

}

func Part2(filename string) int {

	lines := common(filename)

	sum := 0
	for _, line := range lines {

		chars := strings.Split(line, "")

		numbers := []int{}
		for index, char := range chars {
			number, err := strconv.Atoi(char)

			switch {
			case err == nil:
				numbers = append(numbers, number)
			case strings.HasPrefix(line[index:], "one"):
				numbers = append(numbers, 1)
			case strings.HasPrefix(line[index:], "two"):
				numbers = append(numbers, 2)
			case strings.HasPrefix(line[index:], "three"):
				numbers = append(numbers, 3)
			case strings.HasPrefix(line[index:], "four"):
				numbers = append(numbers, 4)
			case strings.HasPrefix(line[index:], "five"):
				numbers = append(numbers, 5)
			case strings.HasPrefix(line[index:], "six"):
				numbers = append(numbers, 6)
			case strings.HasPrefix(line[index:], "seven"):
				numbers = append(numbers, 7)
			case strings.HasPrefix(line[index:], "eight"):
				numbers = append(numbers, 8)
			case strings.HasPrefix(line[index:], "nine"):
				numbers = append(numbers, 9)
			}

		}

		sum = sum + 10*numbers[0] + numbers[len(numbers)-1]
	}

	return sum

}

File: julia_solutions/day01/day01.jl
module Day01

export run

# Function for the problem
function run(input::String)
    lines = readlines(input)
    part_1 = part1(lines)
    part_2 = part2(lines)
    return [part_1, part_2]
end

# Function for part1 of the problem
function part1(lines::Vector{String})
    value = 0
    for line in lines
        numbers = []
        for charac in line
            try
                number = parse(Int, charac)
                append!(numbers, number)
            catch
                continue
            end
        end
        value +=  10*numbers[1] + numbers[end]
    end
    return value
end

function part2(lines::Vector{String})
    value = 0
    for line in lines
        numbers = []
        for (index,charac) in enumerate(line)
            try
                number = parse(Int, charac)
                append!(numbers, number)
            catch
                if startswith(line[index:end], "one")
                    append!(numbers, 1)
                elseif startswith(line[index:end], "two")
                    append!(numbers, 2)
                elseif startswith(line[index:end], "three")
                    append!(numbers, 3)
                elseif startswith(line[index:end], "four")
                    append!(numbers, 4)
                elseif startswith(line[index:end], "five")
                    append!(numbers, 5)
                elseif startswith(line[index:end], "six")
                    append!(numbers, 6)
                elseif startswith(line[index:end], "seven")
                    append!(numbers, 7)
                elseif startswith(line[index:end], "eight")
                    append!(numbers, 8)
                elseif startswith(line[index:end], "nine")
                    append!(numbers, 9)
                end
            end
        end
        value +=  10*numbers[1] + numbers[end]
    end
    return value
end

end